controller/service.go

```diff
@@ -214,3 +257,37 @@ func (c *controller) allocateIPs(key string, svc *v1.Service) ([]net.IP, error)
 	// Okay, in that case just bruteforce across all pools.
 	return c.ips.Allocate(key, serviceIPFamily, k8salloc.Ports(svc), k8salloc.SharingKey(svc), k8salloc.BackendKey(svc))
 }
+
+func (c *controller) getDesiredLbIPs(svc *v1.Service) ([]net.IP, error) {
+	lbIPsStr := svc.Annotations[annotationLoadBalancerIPs]
+	if lbIPsStr == "" {
+		return nil, nil
+	}
+	var lbIPs []net.IP
+	desiredLbIPs := strings.Split(lbIPsStr, ",")
+	for _, desiredLbIP := range desiredLbIPs {
+		ip := net.ParseIP(desiredLbIP)
+		if ip == nil {
+			return nil, fmt.Errorf("invalid %s: %q", annotationLoadBalancerIPs, lbIPsStr)
+		}
+		lbIPs = append(lbIPs, ip)
+	}
+	return lbIPs, nil
+}
+
+func (c *controller) isLbIPsEqual(ipsA []net.IP, ipsB []net.IP) (bool, error) {
+	if len(ipsA) > 2 || len(ipsB) > 2 {
+		return false, fmt.Errorf("unsupported number of load balancer IPs")
+	}
+	if len(ipsA) != len(ipsB) {
+		return false, nil
+	}
+	if len(ipsA) == 2 && (ipsA[0].String() == ipsB[0].String() || ipsA[0].String() == ipsB[1].String()) &&
```

Comment:
Also, you can use the `Equal` method of net.IP, no need to convert to string.

Reply:
ah! correct, anyway this is no more needed due to using `reflect.DeepEqual`